bundles/org.openhab.core/src/main/java/org/openhab/core/common/registry/AbstractRegistry.java

```diff
@@ -244,6 +244,14 @@ public void removed(Provider<E> provider, E element) {
                         element.getClass().getSimpleName(), uid, provider.getClass().getSimpleName());
                 return;
             }
+            Provider<E> elementProvider = elementToProvider.get(existingElement);
+            if (!elementProvider.equals(provider)) {
+                logger.debug(
+                        "Cannot remove \"{}\" with key \"{}\" from provider \"{}\" because the instance in the registry is from provider \"{}\"!",
```

Comment:
```suggestion
                logger.error(
                        "Provider '{}' is not allowed to remove element '{}' with key '{}' from the registry because it was added by provider '{}'.",
```

and adjust the order of objects.

Otherwise LGTM.